src/js/tools/type.js

```diff
@@ -38,13 +44,26 @@ define(function (require, exports, module) {
             firstLaunch = true;
             
         var selectHandler = function () {
+            _moveHandler = function () {
+                var documentStore = this.flux.store("application"),
+                    currentDocument = documentStore.getCurrentDocument();
+
+                this.flux.actions.layers.resetBounds(currentDocument, currentDocument.layers.allSelected);
+            }.bind(this);
+            
+            descriptor.addListener("move", _moveHandler);
+
             if (firstLaunch) {
                 firstLaunch = false;
                 return descriptor.batchPlayObjects([resetObj]);
             }
         };
 
-        Tool.call(this, "typeCreateOrEdit", "Type", "typeCreateOrEditTool", selectHandler);
+        var deselectHandler = function () {
+            descriptor.removeListener("move", _moveHandler);
```

Comment:
This could get leaked on reset. Should we change `actions.tools.onReset` to call the active tool's deselect handler? Or does it already?